dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/merger/MergerFactory.java

```diff
@@ -49,34 +56,51 @@ public void setScopeModel(ScopeModel scopeModel) {
             throw new IllegalArgumentException("returnType is null");
         }
 
-        Merger result;
-        if (returnType.isArray()) {
-            Class type = returnType.getComponentType();
```

Comment:
What is the purpose of removing `returnType.getComponentType()`?

Reply:
one question is why the parameter type of Merger.merge() must be array? If use returnType.getComponentType() in MergerFactory, other non-array types can not exist in MergerFactory. 
for example int[] and int type can not both exist in MergerFactory.

Reply:
> Another problem is MergerFactory caches merger by return type. But there may exist more than one merger with same return type, for example IntFindAnyMerger and IntFindFirstMerger in this pr.